File: source/frontend/src/lib/dto/response/entity/NonFunctionalRequirementResponseDto.ts
import {NonFunctionalRequirement} from "$lib/domain/entity/ar/NonFunctionalRequirement.svelte.js";

export class NonFunctionalRequirementResponseDto {
    id : string = "";
    title : string = "";
}

export function fromNonFunctionalRequirementsResponseDto(nonFunctionalRequirementsResponseDto : NonFunctionalRequirementResponseDto[]) : NonFunctionalRequirement[] {
    return nonFunctionalRequirementsResponseDto.map(fromNonFunctionalRequirementResponseDto);
}

export function fromNonFunctionalRequirementResponseDto(nonFunctionalRequirementResponseDto : NonFunctionalRequirementResponseDto) : NonFunctionalRequirement {
    const nonFunctionalRequirement : NonFunctionalRequirement = NonFunctionalRequirement.create();

    nonFunctionalRequirement.id = nonFunctionalRequirementResponseDto.id;
    nonFunctionalRequirement.title = nonFunctionalRequirementResponseDto.title;

    return nonFunctionalRequirement;
}

File: source/backend/src/main/java/ch/unisg/backend/controller/http/dto/response/ar/NonFunctionalRequirementResponseDto.java
package ch.unisg.backend.controller.http.dto.response.ar;

import ch.unisg.backend.core.domain.entities.classes.ar.NonFunctionalRequirement;

import java.net.URI;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.UUID;
import java.util.stream.Collectors;

public class NonFunctionalRequirementResponseDto {

    private static final String LOCATION = "http://localhost:4000/api/v1/";

    public static URI uri(UUID id) {
        return URI.create(LOCATION + id.toString());
    }

    public static HashMap<String, Object> create(NonFunctionalRequirement nonFunctionalRequirement) {

        HashMap<String, Object> result = new HashMap<>();

        result.put("id", nonFunctionalRequirement.getId().toString());
        result.put("title", nonFunctionalRequirement.getTitle());

        return result;
    }

    public static List<HashMap<String, Object>> toNonFunctionalRequirementResponseDtoList(List<NonFunctionalRequirement> nonFunctionalRequirementList) {
        return nonFunctionalRequirementList.stream().map(NonFunctionalRequirementResponseDto::create).toList();
    }
}
